clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -488,7 +491,11 @@ private void close(KafkaChannel channel) {
         } catch (IOException e) {
             log.error("Exception closing connection to node {}:", channel.id(), e);
         }
-        this.stagedReceives.remove(channel);
+        Deque<NetworkReceive> deque = this.stagedReceives.remove(channel);
+        if (deque != null) {
+            addToCompletedReceives(channel, deque, true);
```

Comment:
This is an interesting change. Can you explain when it can happen that a channel is closed while having a staged receive?